### PR TITLE
feat: add granite bee documentation

### PR DESCRIPTION
**Pull request by grahamwhiteuk**
### Which issue(s) does this pull-request address?

Contributes-to: https://github.com/i-am-bee/internal/issues/19

### Description

Provides accompanying "cookbook" style documentation for the [ibm_granite_bee](https://github.com/i-am-bee/bee-agent-framework/blob/main/examples/agents/ibm_granite_bee.ts) example agent.

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

**Comment by grahamwhiteuk** [comment]
Note to the reviewers, the [formatted version of the Markdown](https://github.com/grahamwhiteuk/bee-agent-framework/blob/feat/ibm-granite-bee-docs/docs/ibm-granite.md) in this PR can be seen in my branch.

File: docs/ibm-granite.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   The default prompt is the question `What is the current weather in London?`
+
+   You can provide other prompts as follows:
+
+   ```shell
+    npm run start examples/agents/ibm_granite_bee.ts <<< 'Who is the president of the USA?'
+    npm run start examples/agents/ibm_granite_bee.ts <<< 'What is the spanish word for dog?'
+    npm run start examples/agents/ibm_granite_bee.ts <<< 'What does the alias command do in the BASH shell?'
+    npm run start examples/agents/ibm_granite_bee.ts <<< "What’s the largest technology company by market capitalization right now?"
+    npm run start examples/agents/ibm_granite_bee.ts <<< "What’s the weather going to be like tomorrow in Sydney?"
```
Comment (Tomas2D) [inline docs/ibm-granite.md:84]: We are using `yarn` in this project. Change to `yarn start examples/...`

File: docs/ibm-granite.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+> More detailed installation instructions are available for step 1 in the [Location Installation](https://github.com/i-am-bee/bee-agent-framework?tab=readme-ov-file#local-installation) section of our main README
+
+1. [Clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) the Bee Agent Framework git repository by running the following command:
+
+   ```shell
+   git clone https://github.com/i-am-bee/bee-agent-framework.git
+   yarn install --immutable
+   ```
+
+1. Copy the [.env.template](/.env.template) file to a new file called `.env`
```
Comment (Tomas2D) [inline docs/ibm-granite.md:43]: This is part of `README.md`, but this one is also not complete (The NodeJS setup is missing).

File: docs/ibm-granite.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Fill out the required details within the `.env` file, as follows
+
+   1. For watsonx (replace the values shown in braces):
+
+      ```ini
```
Comment (Tomas2D) [inline docs/ibm-granite.md:49]: I think it should be ```.env

Comment (grahamwhiteuk) [reply docs/ibm-granite.md]: Resolved in [0b85aab](https://github.com/grahamwhiteuk/bee-agent-framework/commit/0b85aabf1dfcabe29bdf9dc6529328c8d74e9afc)

File: docs/ibm-granite.md (hunk 2 above)
Comment (grahamwhiteuk) [reply docs/ibm-granite.md]: Resolved by 7bad13c

File: docs/ibm-granite.md (hunk 1 above)
Comment (grahamwhiteuk) [reply docs/ibm-granite.md]: Good spot!  Resolved in [e808435](https://github.com/i-am-bee/bee-agent-framework/pull/229/commits/e8084350cf0daa37ddf5423d2f65cd0d2b58c887)

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**